src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -108,6 +108,9 @@ internal static SyntaxNode TryGetCorrespondingLambdaBody(SyntaxNode oldBody, Syn
                 case SyntaxKind.LocalFunctionStatement:
                     return GetLocalFunctionBody((LocalFunctionStatementSyntax)newLambda);
 
+                case SyntaxKind.ArrowExpressionClause:
```

Comment:
>case SyntaxKind.ArrowExpressionClause: [](start = 16, length = 38)

This doesn't look right. Local Functions should be represented by the LocalFunctionStatementSyntax, not ArrowExpression. How do we get here?